### ISSUE TITLE
[a11y]: Progress Indicator - Default - Current Step should be disabled since it is active

### ISSUE THREAD

**Issue by guidari** [carbon-design-system/carbon#16596]
### Package

@carbon/react

### Browser

Firefox

### Package version

v1.58.0

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

Current step is not disabled though it is active status which misleads the SR users
it is a only a observation from Raghu.

### Recommended Fix
Current Step needs to be disabled

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-progressbar--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by heloiselui** [comment]
I tested this component and it seems the issue has been resolved. I also asked Raghu from our accessibility team to test it, and it was confirmed that the component is working as expected.

Raghu's observation: 

> Fixed - Based on the interactive variant (since default variant does not have any active status)
